tests/foreman/ui/test_rhai.py

```diff
@@ -0,0 +1,166 @@
+        lifecycle_env = entities.LifecycleEnvironment(
+            name='rhai_env_{0}'.format(gen_string("alpha", 6)),
+            organization=org.id
+        ).create()
+
+        # Upload manifest
+        manifest_path = manifests.clone()
+        org.upload_manifest(path=manifest_path)
+
+        # Create a custom product for RHAI
+        custom_product = entities.Product(
+            name='rhai_product_{0}'.format(gen_string("alpha", 6)),
+            organization=org
+        ).create()
+
+        # Add insights repo to the custom product
+        custom_repo = entities.Repository(
+            name='insights_repo_{0}'.format(gen_string("alpha", 6)),
+            product=custom_product, content_type=u'yum',
+            url=u'http://access.redhat.com/insights/repo/6/'
+        ).create()
+
+        # Sync custom repository
+        custom_repo.sync()
+
+        # Create content view
+        content_view = entities.ContentView(
+            name='rhai_cv_{0}'.format(gen_string("alpha", 6)),
+            organization=org
+        ).create()
+
+        # Associate custom repository to new content view
+        content_view.set_repository_ids([custom_repo.id])
+
+        # Publish content view
+        content_view.publish()
+
+        # Promote content view to lifecycle_env
+        versions = content_view.read_json()['versions']
+        self.assertEqual(len(versions), 1)
+        entities.ContentViewVersion(
+            id=versions[0]['id']
+        ).promote(lifecycle_env.id)
+
+        # Create activation key
+        activation_key = entities.ActivationKey(
+            auto_attach=True,
+            name=activation_key_name,
+            environment=lifecycle_env.id,
+            organization=org.id,
+            content_view=content_view.id,
+        ).create()
+
+        # Walk through the list of subscriptions.
+        # Find the "Red Hat Employee Subscription" and attach it to the
+        # recently-created activation key.
+        for subscription in org.subscriptions():
+            if subscription['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
+                # 'quantity' must be 1, not subscription['quantity']. Greater
+                # values produce this error: "RuntimeError: Error: Only pools
+                # with multi-entitlement product subscriptions can be added to
+                # the activation key with a quantity greater than one."
+                activation_key.add_subscriptions({
+                    'quantity': 1,
+                    'subscription_id': subscription['id'],
+                })
+
+            # Attach custom product subscription to the activation key
+            if subscription['product_name'] == custom_product.name:
+                activation_key.add_subscriptions({
+                    'quantity': 1,
+                    'subscription_id': subscription['id'],
+                })
+                break
+
+        # Create VM
+        with VirtualMachine(distro='rhel66') as vm:
```

Comment:
I was testing some register stuff and using rhel65 worked better than using rhel66. I recommend you to change only if your test does not pass. Please paste the test results for running this code.